crates/edr_napi/test/issues.ts

```diff
@@ -149,4 +151,125 @@ describe("Provider", () => {
+                32, 32, 32, 32, 125, 44, 10, 32, 32, 32, 32, 34, 115, 111, 117,
+                114, 99, 101, 115, 34, 58, 32, 123, 10, 32, 32, 32, 32, 32, 32,
+                34, 99, 111, 110, 116, 114, 97, 99, 116, 115, 47, 67, 111, 117,
+                110, 116, 101, 114, 46, 115, 111, 108, 34, 58, 32, 123, 10, 32,
+                32, 32, 32, 32, 32, 32, 32, 34, 99, 111, 110, 116, 101, 110,
+                116, 34, 58, 32, 34, 47, 47, 32, 83, 80, 68, 88, 45, 76, 105,
+                99, 101, 110, 115, 101, 45, 73, 100, 101, 110, 116, 105, 102,
+                105, 101, 114, 58, 32, 85, 78, 76, 73, 67, 69, 78, 83, 69, 68,
+                92, 110, 112, 114, 97, 103, 109, 97, 32, 115, 111, 108, 105,
+                100, 105, 116, 121, 32, 94, 48, 46, 56, 46, 50, 56, 59, 92, 110,
+                92, 110, 99, 111, 110, 116, 114, 97, 99, 116, 32, 67, 111, 117,
+                110, 116, 101, 114, 32, 123, 92, 110, 32, 32, 117, 105, 110,
+                116, 32, 112, 117, 98, 108, 105, 99, 32, 120, 59, 92, 110, 92,
+                110, 32, 32, 101, 118, 101, 110, 116, 32, 73, 110, 99, 114, 101,
+                109, 101, 110, 116, 40, 117, 105, 110, 116, 32, 98, 121, 41, 59,
+                92, 110, 92, 110, 32, 32, 102, 117, 110, 99, 116, 105, 111, 110,
+                32, 105, 110, 99, 40, 41, 32, 112, 117, 98, 108, 105, 99, 32,
+                123, 92, 110, 32, 32, 32, 32, 120, 43, 43, 59, 92, 110, 32, 32,
+                32, 32, 101, 109, 105, 116, 32, 73, 110, 99, 114, 101, 109, 101,
+                110, 116, 40, 49, 41, 59, 92, 110, 32, 32, 125, 92, 110, 92,
+                110, 32, 32, 102, 117, 110, 99, 116, 105, 111, 110, 32, 105,
+                110, 99, 66, 121, 40, 117, 105, 110, 116, 32, 98, 121, 41, 32,
+                112, 117, 98, 108, 105, 99, 32, 123, 92, 110, 32, 32, 32, 32,
+                114, 101, 113, 117, 105, 114, 101, 40, 98, 121, 32, 62, 32, 48,
+                44, 32, 92, 34, 105, 110, 99, 66, 121, 58, 32, 105, 110, 99,
+                114, 101, 109, 101, 110, 116, 32, 115, 104, 111, 117, 108, 100,
+                32, 98, 101, 32, 112, 111, 115, 105, 116, 105, 118, 101, 92, 34,
+                41, 59, 92, 110, 32, 32, 32, 32, 120, 32, 43, 61, 32, 98, 121,
+                59, 92, 110, 32, 32, 32, 32, 101, 109, 105, 116, 32, 73, 110,
+                99, 114, 101, 109, 101, 110, 116, 40, 98, 121, 41, 59, 92, 110,
+                32, 32, 125, 92, 110, 125, 92, 110, 92, 110, 34, 10, 32, 32, 32,
+                32, 32, 32, 125, 10, 32, 32, 32, 32, 125, 10, 32, 32, 125, 10,
+                125,
+              ]),
+              output: Uint8Array.from([]),
+            },
+          ],
+          ignoreContracts: false,
+        }
+      );
+
+      assert.fail("Expected an error but none was thrown");
+    } catch (error) {
+      assert.equal(
+        (error as any).message,
+        "Failed to parse build info: EOF while parsing a value at line 1 column 0"
+      );
```

Comment:
@fvictorio I tried to get this to work with chai's `assert.isRejected` but it would still throw an error the moment that `context.createProvider` was called. The thrown error would also point to the line of `context.createProvider` as causing the thrown error.

That was surprising to me as I'd expect the `Promise` to not be linkable to its original callsite. I'd also expect the error to nicely be bubbled up to when I call `const provider = await providerPromise;`.

Is my expectation of Javascript promises incorrect?

Reply:
chai is notoriously bad, so it may be its fault. I you share the code, I can take a look.

Reply:
It sounds and looks like the error is happening synchronously. So even though `createContext` returns a `Promise` the error is happening before the call to `spawn_blocking`.